### PR TITLE
Fix Automatic Download of Pretrained Weights in DETR

### LINKED ISSUE #15764: Resnet weights should not be downloaded when building DETR by from_pretrained
## Describe the bug
Hello, the building process of DETR in transformers uses timm to construct a resnet backbone.
```
backbone = create_model(name, pretrained=True, features_only=True, out_indices=(1, 2, 3, 4), **kwargs)
```
However, `pretrained=True` should only be enabled when building an initial DETR. When building DETR with `from_pretrained`, the pretrained weights of Resnet are redundant. If there is a lack of internet connection or a poor network environment, this can cause the code to get stuck even if the model is building from local files.

## Environment info

- `transformers` version: 4.15.0
- Platform: Linux-5.11.0-46-generic-x86_64-with-glibc2.17
- Python version: 3.8.10
- PyTorch version (GPU?): 1.8.2 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

Models:
DETR

### PR DESCRIPTION
**Pull request by AnugunjNaman**
# What does this PR do?
Fixes #15764

### CODE REVIEW

**Comment by AnugunjNaman** [comment]
@NielsRogge

File: src/transformers/models/detr/modeling_detr.py [hunk 1]
```diff
@@ -1234,6 +1236,9 @@ def forward(
         >>> image = Image.open(requests.get(url, stream=True).raw)
 
         >>> feature_extractor = DetrFeatureExtractor.from_pretrained("facebook/detr-resnet-50")
+        >>> # model use pretrained backbone weights by default
+        >>> # to prevent this set use_pretrained_backbone = False
+        >>> # model = DetrModel.from_pretrained("facebook/detr-resnet-50", use_pretrained_backbone = False)
```
[the remark this answers is not in the document]
Comment (AnugunjNaman) [reply src/transformers/models/detr/modeling_detr.py]: Done. Works like you asked.

**Review by NielsRogge — APPROVED** [review]
Thanks for adding! We could perhaps add a small code example snippet in detr.mdx.

**Comment by AnugunjNaman** [comment]
@NielsRogge Any Update here?

File: docs/source/en/model_doc/detr.mdx [hunk 2]
```diff
@@ -112,6 +112,20 @@ Tips:
...
+      Option 1: instantiate DETR with pre-trained weights for entire model
+          model = DetrForObjectDetection.from_pretrained("facebook/resnet-50")
+
+      Option 2: instantiate DETR with randomly initialized weights for Transformer, but pre-trained weights for backbone
+          config = DetrConfig()
+          model = DetrForObjectDetection(config)
+
+      Option 3: instantiate DETR with randomly initialized weights for backbone + Transformer
+          config = DetrConfig(use_pretrained_backbone=False)
+          model = DetrForObjectDetection(config)
```
Comment (NielsRogge) [inline docs/source/en/model_doc/detr.mdx:128]:
```suggestion
      ```
      from transformers import DetrConfig

      Option 1: instantiate DETR with pre-trained weights for entire model
          model = DetrForObjectDetection.from_pretrained("facebook/resnet-50")

      Option 2: instantiate DETR with randomly initialized weights for Transformer, but pre-trained weights for backbone
          config = DetrConfig()
          model = DetrForObjectDetection(config)

      Option 3: instantiate DETR with randomly initialized weights for backbone + Transformer
          config = DetrConfig(use_pretrained_backbone=False)
          model = DetrForObjectDetection(config)
      ```
```
This won't properly render if you don't add the code statements.

Comment (NielsRogge) [reply docs/source/en/model_doc/detr.mdx:128]: Oh I think you need to do ```python for the first statement

Comment (AnugunjNaman) [reply docs/source/en/model_doc/detr.mdx:128]: Did exactly that https://github.com/huggingface/transformers/pull/17712/commits/66eae21bdf95db6edaf2274f3449a865c80552f8

Comment (NielsRogge) [reply docs/source/en/model_doc/detr.mdx:128]: cc'ing @sgugger here

File: src/transformers/models/detr/configuration_detr.py [hunk 3]
```diff
@@ -82,6 +82,8 @@ class DetrConfig(PretrainedConfig):
             Name of convolutional backbone to use. Supports any convolutional backbone from the timm package. For a
             list of all available models, see [this
             page](https://rwightman.github.io/pytorch-image-models/#load-a-pretrained-model).
+        use_pretrained_backbone ('bool', *optional*, defaults to `True`):
```
Comment (sgugger) [inline src/transformers/models/detr/configuration_detr.py:85]:
```suggestion
        use_pretrained_backbone (`bool`, *optional*, defaults to `True`):
```

**Review by sgugger — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**